File: server/scripts/check_db.mjs
import { db } from '../src/db.js';

console.log('=== questions 表结构 ===');
console.log(JSON.stringify(db.prepare('PRAGMA table_info(questions)').all(), null, 1));
console.log('=== 题目总数 ===');
console.log(JSON.stringify(db.prepare('SELECT subject, COUNT(*) c FROM questions GROUP BY subject').all()));
console.log('=== 无章节题目 ===');
console.log(JSON.stringify(db.prepare("SELECT COUNT(*) c FROM questions WHERE chapter = '' OR chapter IS NULL").get()));
console.log('=== 无答案题目 ===');
console.log(JSON.stringify(db.prepare("SELECT COUNT(*) c FROM questions WHERE answer = '' OR answer IS NULL").get()));
console.log('=== 各来源统计 ===');
console.log(JSON.stringify(db.prepare('SELECT source, COUNT(*) c FROM questions GROUP BY source ORDER BY c DESC LIMIT 20').all(), null, 1));

File: check_db.py
import sqlite3
conn = sqlite3.connect(r'E:\saixt\server\data\saixt.db')
cur = conn.cursor()

print('=== 当前题库统计 ===')
cur.execute("SELECT subject, chapter, COUNT(*) FROM questions GROUP BY subject, chapter ORDER BY subject, chapter")
for r in cur.fetchall():
    print(f'  {r[0]} | {r[1]} | {r[2]}题')

print('\n=== 来源统计 ===')
cur.execute("SELECT source, COUNT(*) FROM questions GROUP BY source ORDER BY COUNT(*) DESC LIMIT 20")
for r in cur.fetchall():
    print(f'  {r[0]}: {r[1]}题')

print('\n=== 各科目总数 ===')
cur.execute("SELECT subject, COUNT(*) FROM questions GROUP BY subject ORDER BY COUNT(*) DESC")
for r in cur.fetchall():
    print(f'  {r[0]}: {r[1]}题')

conn.close()
